### PR TITLE
Communicator Texting And UI Overhaul

### PR DESCRIPTION
**Pull request by SinTwo**
Going to go ahead and do a PR for this now before I move to the next step.  This PR consists of updating the communicator.dm and communicator.tmpl to be ready for modules like on the current PDA's.  This is intended to allow communicators to eventually replace PDA's and in the process a massive overhaul of what was PDA code.  To start with messaging can now be done using the EPv2 system set up by Neerti and has been adapted to now allow both calling and texting (im'ing, messaging) from communicators.

General UI for communicators has been overhauled to accommodate more modules as they are implemented; however, expect a second overhaul in the future after (/if) I can successfully update from HTML4(?) and CSS2(?) to the current standard of HTML5 and CSS3.  That update is to come and will allow much greater control over styling and hopefully increase performance of NanoUI and such.  Enough about that, here's what you need to know.

**TL;DR**

- Texting on communicator
- Fancy new UI on communicator
- Eventual updates for modules (/apps) on communicators
- Pictures vv

Home:
ss 2016-05-16 at 02 53 07

Dialing:
ss 2016-05-16 at 02 53 11

Contacts List:
ss 2016-05-16 at 02 53 23

Message Threads:
ss 2016-05-16 at 02 53 29

Message:
ss 2016-05-16 at 02 56 02

Note Keeper:
ss 2016-05-16 at 02 56 36

Settings:
ss 2016-05-16 at 02 56 42

### CODE REVIEW

**Comment by Arokha** [comment]
Slick. Please.

**Comment by Datraen** [comment]
:+1:

**Comment by EmperorJon** [comment]
The notepad still has the PDA text intro there, as a reminder :P

**Comment by SinTwo** [comment]
@EmperorJon *shrug.
I'm going to change the formating for this in the coming days as I work with the new things I get with CSS3 and HTML5. PR for that coming soon.

File: code/datums/EPv2.dm [hunk 1]
```diff
@@ -105,23 +105,23 @@ var/global/list/all_exonet_connections = list()
 // Proc: send_message()
 // Parameters: 2 (target_address - the desired address to send the message to, message - the message to send)
 // Description: Sends the message to target_address, by calling receive_message() on the desired datum.
-/datum/exonet_protocol/proc/send_message(var/target_address, var/message)
+/datum/exonet_protocol/proc/send_message(var/target_address, var/message, var/text)
```
Comment (Neerti) [inline code/datums/EPv2.dm]: If you're going to add another argument, can you update the comment above it telling everyone what it does?
